### PR TITLE
Database revamp

### PR DESCRIPTION
**Pull request by tvheems**
Made the database and graph structure cache for faster loading.
Made cold load faster.
Changed database to a text file.

### CODE REVIEW

File: src/main/java/datastructure/NodeGraph.java [hunk 1]
```diff
@@ -34,23 +39,31 @@ public NodeGraph() {
...
         }
 
+        size++;
         int[] temp = nodes.get(id).getIncomingEdges();
         nodes.set(id, node);
         nodes.get(id).setIncomingEdges(temp);
-        segments.addSegment(id, segment);
+    }
+
+    public void addNodeCache(int id, Node node) {
```
Comment (blazeuser) [inline src/main/java/datastructure/NodeGraph.java]: Add javadoc.

Comment (DexterQvL) [inline src/main/java/datastructure/NodeGraph.java]: Javadoc has been added.

File: src/main/java/datastructure/NodeTest.java [hunk 2]
```diff
@@ -1,45 +1,48 @@
...
         assertTrue(node.equals(new Node(0, new int[0], new int[0])));
     }
 
-    @org.junit.jupiter.api.Test
+    @org.junit.Test
     void equalsNull() {
         assertFalse(node.equals(null));
     }
+
 }
```
Comment (DexterQvL) [inline src/main/java/datastructure/NodeTest.java]: I looked through the classes again, and fixed all checkstyle warnings I found.

File: src/main/java/parsing/Parser.java [hunk 3]
```diff
@@ -93,6 +146,57 @@ public NodeGraph parse(final String filename, NodeGraph graph) {
             e.printStackTrace();
         }
 
+
         return graph;
     }
```
Comment (MStraatman) [inline src/main/java/parsing/Parser.java]: Method length...

**Pull request merged**